htdocs/takepos/invoice.php

```diff
@@ -309,7 +310,7 @@ function fail($message)
 				$payment->amounts[$invoice->id] = $amountofpayment;
 				// If user has not used change control, add total invoice payment
 				// Or if user has used change control and the amount of payment is higher than remain to pay, add the remain to pay
-				if ($amountofpayment <= 0 || $amountofpayment > $remaintopay) {
+				if (abs($amountofpayment) <= 0 || abs($amountofpayment) > abs($remaintopay)) {
```

Comment:
abs(x) is <= 0 only if amountofpayment is 0.
Surely not what the result you expect.